pkg/infra/httpclient/httpclientprovider/user_agent_middleware.go

```diff
@@ -0,0 +1,27 @@
+package httpclientprovider
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+)
+
+// SetUserAgentMiddlewareName the middleware name used by SetUserAgentMiddleware.
+const SetUserAgentMiddlewareName = "user-agent"
+
+// SetUserAgentMiddleware middleware that sets the HTTP header User-Agent on the outgoing request.
```

Comment:
Nit, but the following reads more like a real sentence:
```suggestion
// SetUserAgentMiddleware is middleware that sets the HTTP header User-Agent on the outgoing request.
```